src/lib/middleware/featureFlagMiddleware.ts

```diff
@@ -0,0 +1,48 @@
+import { NextFunction } from "express"
+import { ArtsyRequest, ArtsyResponse } from "./artsyExpress"
+import {
+  createFeatureFlagService,
+  FeatureFlagService,
+} from "lib/featureFlags/featureFlagService"
+
+export function featureFlagMiddleware(serviceType: symbol) {
+  let service
+  return (req: ArtsyRequest, res: ArtsyResponse, next: NextFunction) => {
+    new Promise<FeatureFlagService>(async (resolve, reject) => {
+      if (service) {
+        resolve(service)
+      } else {
+        try {
+          service = await createFeatureFlagService(serviceType)
+          resolve(service)
+        } catch {
+          reject("An unknown error occurred while creating the flag service.")
+        }
+      }
+    })
+      .then(service => {
+        // Create feature flag context per request
+        const featureFlagContext = {
+          userId: res.locals.user ? res.locals.user.id : null,
+          sessionId: res.locals.sd.SESSION_ID,
+        }
+
+        // Get features and move them to sharify
+        const flags = service.getFeatures()
+        res.locals.sd.featureFlags = {}
+        if (flags) {
+          for (let flag of flags) {
+            res.locals.sd.featureFlags[flag] = service.enabled(
+              flag,
+              featureFlagContext
+            )
+          }
+        }
+        next()
+      })
+      .catch(error => {
+        console.error(error)
```

Comment:
```tsx
console.error("[Force] Error creating feature flag service:", error)
```

Reply:
As a side note, this mixture of `async / await` and vanilla promises (`.then/.catch`) is a bit confusing. Best to stick with `async / await` everywhere per current JS standards.